src/node_perf.cc

```diff
@@ -440,8 +432,6 @@ void Initialize(Local<Object> target,
                             env->constants_string(),
                             constants,
                             attr).ToChecked();
-
-  SetupGarbageCollectionTracking(env);
```

Comment:
I would actually prefer for this to stay here. It makes the code more local and more obvious to me, and only adds the callbacks when they are actually used.

Reply:
@addaleax this is currently unconditionally called when the isolate goes through bootstrapping - maybe this should be delayed until the first time the user adds an `gc` performance observer..

Reply:
@joyeecheung If that’s feasible, it sounds good to me, but I’m more worried about code organization than performance.

Also, I’m just noticing that we never remove these listeners … that’s a bug that we should fix if we can?

Reply:
See https://github.com/nodejs/node/pull/25647 – I’m happy to close it if you want to do sth in this PR :)

Reply:
@addaleax Thanks, I think I'd prefer moving `SetupGarbageCollectionTracking` into a binding method and call it from JS when it's actually needed.